vertical-pod-autoscaler/pkg/admission-controller/main.go

```diff
@@ -102,11 +104,16 @@ func main() {
 	if err != nil {
 		klog.Fatalf("Unable to get hostname: %v", err)
 	}
+
+	statusNamespace := status.AdmissionControllerStatusNamespace
+	if *vpaObjectNamespace != "" {
+		statusNamespace = *vpaObjectNamespace
```

Comment:
This makes me think we should call it vpaControlledNamespace.

Do you plan to limit VPA rbac to only the chosen namespace and this is why you also need to change the namespace for ac status object?

Reply:
maybe this needs to be a seperate flag?

Because right now I'm running vpa in the **x** namespace and it manages VPA objects in **x** namespace and it works fine for me, but I think a decent use case is to run vpa in **x** namespace but make it watch **y** namepsace. And  **x* can be configured via [fieldRef](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/):

```
        args: 
         --vpa-object-namespace=y
         --vpa-controller-namespace=$(VPA_CONTROLLER_NAMESPACE)
        env:
        - name: VPA_CONTROLLER_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
```

What do you think?

Reply:
Yeah this approach looks good to me

Reply:
Actually, this already works like this, no? We have 
namespace          = os.Getenv("NAMESPACE") 
as a flag (since I think k8s sets those env variables by default)

Reply:
I've looked at a bunch of prods in my cluster, couldn't find env var "NAMESPACE".

Looked on internet for the answers and I believe, by default it's not exposed, you have to setup it via Downward API: https://stackoverflow.com/a/52681154

Docs also seem to point to Downward API:
https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/
https://kubernetes.io/docs/concepts/containers/container-environment/#container-information